src/pedal/config.rs

```diff
@@ -0,0 +1,50 @@
+use crate::{pedal, Pedal as PedalTrait, Result};
+use cpal::StreamConfig;
+use serde::Deserialize;
+use Pedal::*;
+
+#[derive(Debug, Deserialize)]
+pub struct Config {
+    pedals: Vec<Pedal>,
+}
+
+impl Config {
+    pub fn from(yaml: &str) -> Result<Config> {
+        Ok(serde_yaml::from_str(yaml)?)
+    }
+
+    pub fn default() -> Self {
+        Self {
+            pedals: vec![Transparent],
+        }
+    }
+
+    pub fn to_pedals(&self, stream_config: &StreamConfig) -> Result<Vec<pedal::Boxed>> {
+        self.pedals
+            .iter()
+            .map(|pedal| pedal.to_pedal(stream_config))
+            .collect()
+    }
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(tag = "type")]
+enum Pedal {
+    Transparent,
+    Delay(DelayConfig),
+}
+
+impl Pedal {
+    fn to_pedal(&self, stream_config: &StreamConfig) -> Result<pedal::Boxed> {
+        Ok(match self {
+            Transparent => pedal::Transparent::new().boxed(),
+            Delay(delay_config) => pedal::Delay::new(stream_config, delay_config)?.boxed(),
```

Comment:
Using the `pedal::` namespacing here since there are overlapping names between the config enum cases and the pedal structs.